### ISSUE TITLE
Multiple Time.zone support

### ISSUE THREAD

**Issue by Nerian** [code-and-effect/effective_datatables#107]
Hi,

I am implementing reservation management now and I just noticed some weird behaviour when datatable is showing DateTime fields. The time (hours) part is showing correctly when the table is first initialised, but then when a filter is used the table actually shows times as if the timezone was UTC. 

Which makes total sense since the Datatable Filter part is handled from a controller in EffectiveDatatable.

In my controllers this bit sets the Time.zone for the current account:

```ruby
class ApplicationController < ActionController::Base
  around_action :set_time_zone, if: :current_account

  def set_time_zone
    Time.use_zone(current_account.time_zone) do
      yield
    end
  end

end
```

But that won't effect EffectiveDatatable::DatatablesController.

I was wondering how do you handle different time zones when using EffectiveDatatable.

Thanks

**Comment by Nerian** [comment]
An idea:

Use something similar to what you do in `EffectiveDatatables.authorization_method`. 

If I could have a method that gets called before `DatatablesController#Show` then I can put my Time.zone handler there – this could potentially be used for something too.

**Comment by Nerian** [comment]
Actually I see that DatatablesController inherits from ApplicationController. So why wouldn't things just work O-o.

**Comment by matt-riemer** [comment]
Yea, great questions:

1.) Datatables doesn't do anything with timezones itself.  

https://github.com/code-and-effect/effective_datatables/blob/master/app/models/effective/effective_datatable/format.rb#L82

2.) I do kind of enforce the %Y-%m-%d format so that the searching will work correctly.  You can just type "2019-04" in a date filter box, and it'll pull all the April 2019 records.

3.) Rails and timezones suck.  I generally use two approaches:

a.) I set my rails server config.time_zone to be "MST (Mountain Standard Time)" (that's the zone I'm in) so then all `.strftime` and date displays, are in that MST time zone. 

b.) Some sites I need user specific time zone display.  So I do something very similar to the controller code you posted above.  This has the effect of displaying the time zone as per the configured value.

You could do something like:

`around_action :set_time_zone, if: -> { params['controller'] == 'effective/datatables' && params['action'] == show }`

if you wanted to only affect the datatatables action.

https://github.com/code-and-effect/effective_datatables/blob/master/app/controllers/effective/datatables_controller.rb

**Comment by matt-riemer** [comment]
You could also do a custom column maybe

`col(:my_date) { |date| date.in_time_zone(current_user.time_zone).strftime('%F') }`

**Comment by Nerian** [comment]
Yup, I am using a custom column.

````

col :date, label: I18n.t('general.date') do |reservation|
  logger.debug Time.zone
  "#{l(reservation.date_start, format: :short)} <br> #{l(reservation.date_end, format: :short)}"
end
````

application..rb
```
class Application < Rails::Application
  config.time_zone = 'UTC' 
end
```

On the first render, the logger shows  (GMT+01:00) Madrid. Which is correct for the user.

When I apply a filter, or order, or do anything that will get for the Datatable controller then it uses an incorrect zone:  (GMT+00:00) UTC.  Which is what I set as the application default

**Comment by Nerian** [comment]
I am going to use the demo app and see if I can reproduce this there.

**Comment by Nerian** [comment]
After 1 hour of debugging, and finally monkeypatching the `zone=()` method in ActiveSupport to see how that values were changing around, I noticed that Time.zone was not being set all when doing a remote operation with the table. And then I saw where the error was:  Between the desk and screen.

This was the around filter that I had set:
```
around_action :set_time_zone, if: :current_account
```

That `current_account` is going to return `nil` when within the EffectiveDatatable Controller because it needs the `account_id` in the route. So the filter is not going to act and thereby `Time.zone` would not be set. And that explains it all.

😭 🔫 

~~FUuuuuuuuuuuuuuck~~  Oh blimey, what an unfortunate miss.
